db_migration.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+import psycopg2
+
+from alembic.command import stamp, upgrade
+from alembic.config import Config
+
+
+def migrate(db_url: str):
+    """Ensure the alembic migration state is up-to-date.
+    If the database table "libraries" has not been created yet, we can assume this is a new deployment.
+    Else, we can assume this database should attempt an upgrade to the latest version, if the DB
+    is already at the latest version, alembic will ignore the upgrade command.
+
+    Note: This function must be run before the SQLAlchemy session is initialized.
+    """
+    log = logging.getLogger(__name__)
+    logging.basicConfig()
+    log.setLevel(logging.INFO)
+
+    # Find the 'libraries' table
+    conn = psycopg2.connect(db_url)
+    cursor = conn.cursor()
+    cursor.execute("SELECT * FROM pg_catalog.pg_tables where tablename='libraries';")
+    table_row = cursor.fetchone()
+    cursor.close()
+    conn.close()
```

Comment:
Yes, in the case where we have an existing deployment, we would see this behaviour, in this case only the 'head' needs to be stamped. 
I chose to use an application table instead of an alembic table so we can keep the validation application dependant, and not alembic dependant (eg. they change the name of the `stamp` table)